perfkitbenchmarker/benchmarks/hadoop_terasort_benchmark.py

```diff
@@ -36,21 +36,23 @@
 FLAGS = flags.FLAGS
 
 BENCHMARK_INFO = {'name': 'hadoop_terasort',
-                  'description': 'Runs Terasort',
+                  'description': 'Runs Terasort. Control the number of VMs '
+                  'with --num_vms.',
                   'scratch_disk': True,
-                  'num_machines': 9}
+                  'num_machines': None}
+DEFAULT_VM_COUNT = 9
 
-HADOOP_VERSION = '2.5.2'
-HADOOP_URL = ('http://apache.mirrors.tds.net/hadoop/common/hadoop-%s/'
-              'hadoop-%s.tar.gz') % (HADOOP_VERSION, HADOOP_VERSION)
-
-DATA_FILES = ['core-site.xml.j2', 'yarn-site.xml.j2', 'hdfs-site.xml',
-              'mapred-site.xml', 'hadoop-env.sh', 'slaves.j2']
 NUM_BYTES_PER_ROW = 100
+NUM_MB_PER_ROW = NUM_BYTES_PER_ROW / (1024.0 ** 2)
 
 
 def GetInfo():
-  return BENCHMARK_INFO
+  info = BENCHMARK_INFO.copy()
+  if FLAGS.num_vms == 1:  # Unchanged from default
+    info['num_machines'] = 9
```

Comment:
This seems too magical - if it can't usefully run with the default one VM, I think it would be preferable to refuse to run as opposed to silently reinterpreting the number. If the intent is just to make this configurable, a benchmark-specific flag such as --hadoop-terasort-num-vms seems cleaner.

(Also, that '9' should be DEFAULT_VM_COUNT.)